### PR TITLE
delegate lmultiply to multiply_log; vectorize abs

### LINKED ISSUE #2595: abs vectorization, lmultiply/multiply_log in math lib
## Description

The fixes are so that we don't have to hack `stanc3` to generate code.  

1.  implement `lmultiply` in the math lib by delegating to `multiply_log`
2. implement vectorized form of `abs` for continuous values by delegating to `fabs`
3. implement vectorized integer array version of `abs`

See https://github.com/stan-dev/stanc3/issues/475 for further discussion.

#### Current Version:
v4.1.0

### PR DESCRIPTION
**Pull request by bob-carpenter**
## Summary

Add `lmultiply` function that parallels `multiply_log` so that the language doesn't have to rename.  

Add vectorized signatures for `abs()` that deal with integer returns for integer container inputs.

While I was at it, I removed the annoying warnings from using the deprecated shuffle command from the std lib.

Fixes #2595

## Tests

Copied from the `multiply_log` tests.  So really they're just testing the generic instantiation and the OpenCL logic.

## Side Effects

No.

## Release notes

Replace this text with a short note on what will change if this pull request is merged in which case this will be included in the release notes.

## Checklist

- [x] Math issue #2595

- [x] Copyright holder: Simons Foundation

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [ ] OpenCL tests passing
    - I'm submitting as a WIP PR because I don't know how to do this

- [x] the basic tests are passing
    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)
    - [ ] OpenCL tests --- I'm submitting because I don't know how to run these locally
- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

**Comment by bob-carpenter** [comment]
I'm getting email that tests were cancelled:

* Windows RTools 3.5: All jobs were cancelled
* Header checks: All jobs were cancelled

Did someone do that on purpose or is there a problem with the PR?

**Comment by rok-cesnovar** [comment]
> I'm getting email that tests were cancelled:

If anyone (including Stan buildbot) pushes changes while the previous tests with Github Actions are still running, 
we have it set up so that it cancels those active runs. The Github Action tests are free, but its still 
wastefull to keep those runs alive when we know they wont matter in the end. 

In your case here, Stan buildbot pushed formatting changes which is what cancled the runs that started a few minutes ago.
Long story short, ignore those. 

If this annoys you, you can go in your Notification settings on Github (https://github.com/settings/notifications) and select that it does not send you emails:

image

**Comment by bob-carpenter** [comment]
Thanks for the explanation, @rok-cesnovar.

**Comment by bob-carpenter** [comment]
@SteveBronder:  Can you help me with the OpenCL testing requirements?  It's failing without an informative message and I don't know how to fix.  I probably missed a file that was configuring `multiply_log` something or other somewhere when I copied over for lmultiply.  (This took 6 code files and 9 test files to get going for one function!)
